ledger/src/bank_forks_utils.rs

```diff
@@ -294,11 +294,19 @@ fn bank_forks_from_snapshot(
         })?;
         bank
     } else {
-        let bank_snapshot =
-            latest_bank_snapshot.ok_or_else(|| BankForksUtilsError::NoBankSnapshotDirectory {
+        // fastboot from local state
+        let Some(bank_snapshot) = latest_bank_snapshot else {
+            // If we do *not* have a local snapshot to fastboot, then it must be because there was
+            // *not* a loadable local snapshot AND we shall *not* startup from a snapshot archive.
+            assert_eq!(
```

Comment:
what's going on here?

Reply:
I agree with Jeff here, it seems odd to me. We are basically assigning a bool, then having an assert we're assigning it correctly?

Line 250:
```rust
    let will_startup_from_snapshot_archives = match process_options.use_snapshot_archives_at_startup
    {
        UseSnapshotArchivesAtStartup::Always => true,
        UseSnapshotArchivesAtStartup::Never => false,
        UseSnapshotArchivesAtStartup::WhenNewest => latest_bank_snapshot
            .as_ref()
            .map(|bank_snapshot| latest_snapshot_archive_slot > bank_snapshot.slot)
            .unwrap_or(true),
    };
```

```rust
    let bank = if will_startup_from_snapshot_archives {
        // ...stuff
    } else {
            // fastboot from local state
            let Some(bank_snapshot) = latest_bank_snapshot else {
            // If we do *not* have a local snapshot to fastboot, then it must be because there was
            // *not* a loadable local snapshot AND we shall *not* startup from a snapshot archive.
            assert_eq!(
                process_options.use_snapshot_archives_at_startup,
                UseSnapshotArchivesAtStartup::Never,
            );
            return Err(BankForksUtilsError::NoBankSnapshotDirectory {
                flag: use_snapshot_archives_at_startup::cli::LONG_ARG.to_string(),
                value: UseSnapshotArchivesAtStartup::Never.to_string(),
            });
        };
    }
```